moveit_core/trajectory_processing/src/time_optimal_trajectory_generation.cpp

```diff
@@ -106,8 +106,10 @@ class CircularPathSegment : public PathSegment
 
     const Eigen::VectorXd start_direction = (intersection - start).normalized();
     const Eigen::VectorXd end_direction = (end - intersection).normalized();
+    const double start_dot_end = start_direction.dot(end_direction);
 
-    if ((start_direction - end_direction).norm() < 0.000001)
+    // catch division by 0 in computations below
+    if (start_dot_end > 0.999999 || start_dot_end < -0.999999)
```

Comment:
I'm not really familiar with this algorithm so I'm not 100% sure L114-L118 are the correct thing to do in this case. Are you sure of that?

If you are not sure, maybe it would be best to clip start_dot_end in the range [-0.999999, 0.999999], then let the calculations continue below.

start_dot_end = std::max(-0.999999, std::min(start_dot_end, 0.999999));